gitlab-runner/register-runner.sh

```diff
@@ -1,26 +1,43 @@
-#!/bin/bash
+#!/usr/bin/env bash
 
 # gitlab-runner data directory
 DATA_DIR="/etc/gitlab-runner"
 CONFIG_FILE=${CONFIG_FILE:-$DATA_DIR/config.toml}
-# custom certificate authority path
-CA_CERTIFICATES_PATH=${CA_CERTIFICATES_PATH:-$DATA_DIR/certs/ca.crt}
-LOCAL_CA_PATH="/usr/local/share/ca-certificates/ca.crt"
+CACHE_DIR="/cache"
+# Default RUNNER_NAME=HOSTNAME
+RUNNER_NAME=${RUNNER_NAME:-$HOSTNAME}
 
 update_ca() {
   echo "Updating CA certificates..."
   cp "${CA_CERTIFICATES_PATH}" "${LOCAL_CA_PATH}"
   update-ca-certificates --fresh >/dev/null
 }
 
-if [ -f "${CA_CERTIFICATES_PATH}" ]; then
-  # update the ca if the custom ca is different than the current
-  cmp -s "${CA_CERTIFICATES_PATH}" "${LOCAL_CA_PATH}" || update_ca
-fi
-
-gitlab-runner register --non-interactive \
+register() {
+  gitlab-runner register --non-interactive \
   --docker-volumes "/var/run/docker.sock:/var/run/docker.sock" \
-  --docker-volumes "/cache" \
+  --docker-volumes "${CACHE_DIR}"
+}
+
+if [[ -f "$CONFIG_FILE" ]]; then
```

Comment:
I think the outer if-block which tests for file existence can be removed, since the verify command should do the trick.